File: app/src/main/java/com/example/islamic/view/HadithDetailActivity.java
package com.example.islamic.view;

import android.os.Bundle;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.example.islamic.R;
import com.example.islamic.model.Hadith;

public class HadithDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hadith_detail);

        TextView hadithTitle = findViewById(R.id.hadith_detail_title);
        TextView hadithText = findViewById(R.id.hadith_detail_text);
        TextView hadithDescription = findViewById(R.id.hadith_detail_description);

        Hadith hadith = (Hadith) getIntent().getSerializableExtra("hadith_object");

        if (hadith != null) {
            hadithTitle.setText(hadith.getTitle());
            hadithText.setText(hadith.getHadithText());
            hadithDescription.setText(hadith.getDescription());
        }
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/activity/HadithDetailActivity.kt
package com.eslamdev.islamic.presentation.ui.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.eslamdev.islamic.data.model.HadithModel
import com.eslamdev.islamic.databinding.ActivityHadithDetailBinding

class HadithDetailActivity : AppCompatActivity() {

    private lateinit var binding: ActivityHadithDetailBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHadithDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // استقبال البيانات بالمودل الجديد
        @Suppress("DEPRECATION")
        val hadith = intent.getSerializableExtra("hadith_object") as? HadithModel

        if (hadith != null) {
            binding.hadithDetailTitle.text = hadith.title
            binding.hadithDetailText.text = hadith.content

            // لو مفيش شرح، نخفي العنوان بتاعه عشان الشكل العام
            if (hadith.description.isNotBlank()) {
                binding.hadithDetailDescription.text = hadith.description
            } else {
                binding.hadithDetailDescription.text = "لا يوجد شرح متاح حالياً"
            }
        }

        binding.btnBack.setOnClickListener{
            finish()
        }
    }
}